pkg/apis/application/v1alpha1/types.go

```diff
@@ -453,6 +462,21 @@ type AppProjectSpec struct {
 
 	// Description contains optional project description
 	Description string `json:"description,omitempty" protobuf:"bytes,3,opt,name=description"`
+
+	Roles []ProjectRole `json:"roles,omitempty" protobuf:"bytes,4,rep,name=roles"`
+}
+
+// ProjectRole represents a role that has access to a project
+type ProjectRole struct {
+	Name      string     `json:"name" protobuf:"bytes,1,opt,name=name"`
+	Policies  []string   `json:"policies" protobuf:"bytes,2,rep,name=policies"`
```

Comment:
We will want description here for humans to explain the purpose.

Reply:
fixed